module2/projects/coupon-codes-api/index.md

```diff
@@ -0,0 +1,467 @@
+    "type": "coupon",
+    "attributes": {
+      "name": "Buy One Get One 50",
+      "code": "BOGO50",
+      "amount_off": 50,
+      "percent_off": null,
+      "status": "active",
+      "merchant_id": 1,
+      "uses": 1    
+    }
+  }
+}
+
+```
+</section>
+
+<section class="dropdown">
+### 2. Merchant Coupons Index 
+
+Returns all of a merchant's coupons
+
+**Request**
+```bash
+GET /api/v1/merchants/:merchant_id/coupons
+Content-Type: application/json
+Accept: application/json
+```
+
+**Response**
+
+```json
+{
+  "data": [
+    {
+      "id": "1",
+      "type": "coupon",
+      "attributes": {
+        "name": "Buy One Get One 50",
+        "code": "BOGO50",
+        "amount_off": 50,
+        "percent_off": null,
+        "status": "active",
+        "merchant_id": 1,
+        "uses": 1
+      }
+    },
+    {
+      "id": "2",
+      "type": "coupon",
+      "attributes": {
+        "name": "25 Percent Sale",
+        "code": "25PERCENT",
+        "amount_off": null,
+        "percent_off": 25,
+        "status": "inactive",
+        "merchant_id": 1,
+        "uses": 0
+      }
+    },
+    {
+      "id": "3",
+      "type": "coupon",
+      "attributes": {
+        "name": "One More Coupon",
+        "code": "MONEYOFF",
+        "amount_off": 10,
+        "percent_off": null,
+        "status": "active",
+        "merchant_id": 1,
+        "uses": 2
+      }
+    }
+  ]
+}
+```
+</section>
+
+<section class="dropdown">
+### 3. Merchant Coupon Create 
+
+Create a new coupon for a merchant
+
+**Request**
+```
+POST /api/v1/merchants/:merchant_id/coupons
+Content-Type: application/json
+Accept: application/json
+
+{
+  "name": "Buy One Get One 50",
+  "code": "BOGO50",
+  "amount_off": 50,
+  "percent_off": null,
+  "status": "active",
+  "merchant_id": 1
+}
+```
+
+```json
+status: 201
+body:
+
+{
+  "data": {
+    "id": "1",
+    "type": "coupon",
+    "attributes": {
+      "name": "Buy One Get One 50",
+      "code": "BOGO50",
+      "amount_off": 50,
+      "percent_off": null,
+      "status": "active",
+      "merchant_id": 1,
+      "uses": 0
+    }
+  }
+}
+```
+
+
+**Sad Paths to consider:**
+1. This Merchant already has 5 active coupons
+2. Coupon code entered is NOT unique
+
+</section>
+
+<section class="dropdown">
+### 4. Merchant Coupon Deactivate
+
+Updates a coupon from active to inactive
+
+**Request**
+```bash
+PATCH /api/v1/merchants/:merchant_id/coupons/:id
+Content-Type: application/json
+Accept: application/json
+
+{
+  "status": "inactive"
+}
+```
+
+**Response**
+
+```json
+status: 200
+body:
+
+{
+  "data": {
+    "id": "1",
+    "type": "coupon",
+    "attributes": {
+      "name": "Buy One Get One 50",
+      "code": "BOGO50",
+      "amount_off": 50,
+      "percent_off": null,
+      "status": "inactive",
+      "merchant_id": 1,
+      "uses": 1
+    }
+  }
+}
+```
+
+<!-- **Sad Paths to consider:**
+1. A coupon cannot be deactivated if there are any pending invoices with that coupon. -->
```

Comment:
I think this is a good idea for students to consider! Is  it commented out because you took it out?

Reply:
Yes I commented it out because I was grappling with whether the relationships were ones the students could handle since I am not the most familiar with the db structure of this project. If you think it's good, I'll add it back in!

Reply:
I think that might be a bit tricky but definitely doable for them! This is still just multiple one-to-many relationship style joins.